--- a/src/Database/Table.php
+++ b/src/Database/Table.php
@@ -32,6 +32,7 @@ use BerlinDB\Database\Base;
  */
 abstract class Table extends Base {
 
+	protected $site_id = null;
 	/**
 	 * Table name, without the global table prefix.
 	 *
@@ -206,7 +207,7 @@ abstract class Table extends Base {
 		}
 
 		// Update interface for switched site
-		$this->set_db_interface();
+		$this->set_db_interface($site_id);
 	}
 
 	/** Public Helpers ********************************************************/
@@ -824,8 +825,24 @@ abstract class Table extends Base {
 	 *
 	 * @since 1.0.0
 	 */
-	private function set_db_interface() {
+	private function set_db_interface($site_id = null) {
 
+		// Set variables for global tables
+		if ($this->is_global()) {
+			$site_id = 0;
+			$tables  = 'ms_global_tables';
+		} else {
+			$tables = 'tables';
+		}
+
+		// Assign the correct database interface
+		if ( ! empty( $site_id ) ) {
+			$this->db = $GLOBALS['wpdb']->get_blog_prefix( $site_id );
+		} else {
+			$this->db = $GLOBALS['wpdb'];
+		}
+
+		// Cache the table name
 		// Maybe switch database
 		if ( $this->is_global() ) {
 			$this->apply_prefix();