--- a/src/Broker/Cookies.php
+++ b/src/Broker/Cookies.php
@@ -43,7 +43,8 @@ class Cookies implements \ArrayAccess
     /**
      * @inheritDoc
      */
-    public function offsetSet($name, $value)
+	#[\ReturnTypeWillChange]
+	public function offsetSet($name, $value)
     {
         $success = setcookie($name, $value, time() + $this->ttl, $this->path, $this->domain, $this->secure, true);
 
@@ -66,7 +67,8 @@ class Cookies implements \ArrayAccess
     /**
      * @inheritDoc
      */
-    public function offsetGet($name)
+	#[\ReturnTypeWillChange]
+	public function offsetGet($name)
     {
         return $_COOKIE[$name] ?? null;
     }
@@ -74,7 +76,8 @@ class Cookies implements \ArrayAccess
     /**
      * @inheritDoc
      */
-    public function offsetExists($name)
+	#[\ReturnTypeWillChange]
+	public function offsetExists($name)
     {
         return array_key_exists($name, $_COOKIE);
     }
@@ -82,7 +85,8 @@ class Cookies implements \ArrayAccess
     /**
      * @inheritDoc
      */
-    public function offsetUnset($name)
+	#[\ReturnTypeWillChange]
+	public function offsetUnset($name)
     {
         unset($_COOKIE[$name]);
         setcookie($name, '', 1, $this->path, $this->domain, $this->secure, true);